From 5959ff080317d532ddc8831cafc2a947a4014363 Mon Sep 17 00:00:00 2001
From: David Kilp <David.Kilp@crossmatch.com>
Date: Wed, 18 Oct 2017 10:41:49 -0400
Subject: [PATCH 02/18] [PATCH 2/2] Add patch for Bluetooth as specified by
 Silex module. I believe this patch is also safe for the Ti Winlink parts as
 well.

---
 drivers/tty/serial/imx.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index fe070db..59f96ef 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1278,8 +1278,11 @@ static int imx_startup(struct uart_port *port)
 
 	/* Can we enable the DMA support? */
 	if (is_imx6q_uart(sport) && !uart_console(port)
-		&& !sport->dma_is_inited)
+		&& !sport->dma_is_inited &&
+		!sport->port.flags == UPF_LOW_LATENCY)
+	{
 		imx_uart_dma_init(sport);
+	}
 
 	if (sport->dma_is_inited)
 		INIT_WORK(&sport->tsk_dma_tx, dma_tx_work);
-- 
2.9.5

